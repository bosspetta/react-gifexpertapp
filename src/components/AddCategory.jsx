import React, { useState } from 'react'

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('')

    const onInputChange = (event) => {
        setInputValue(event.target.value)
    }

    const onSubmit = (event) => {
        event.preventDefault()
        if (inputValue.trim().length <= 1) return
        // setCategories(prev => [inputValue, ...prev])
        onNewCategory(inputValue.trim())
        setInputValue('')
    }

    return (
        <form onSubmit={onSubmit}>
            <input
                type="text"
                aria-label="Buscar GIF's por categorÃ­a"
                placeholder='Buca un GIF'
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}
