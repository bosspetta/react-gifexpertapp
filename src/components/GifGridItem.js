import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({id, title, url}) => {
  // console.log({id, title, url});
  return (
    <div className="card animate__animated animate__fadeIn">
      <img id={id} src={url} alt={title} />
      <p>
        <small>
          <strong>{title}</strong>
        </small>
      </p>
    </div>
  );
};

GifGridItem.protoTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
