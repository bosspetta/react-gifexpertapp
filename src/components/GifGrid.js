// import React, { useState, useEffect } from 'react';
import React from 'react';
import PropTypes from 'prop-types'

import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

  const { data:images, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className="animate__animated animate__fadeIn">{category}</h3>
      {loading && (
        <p className="animate__animated animate__flash">
          <img
            src="https://www.ipgvip.com/wp-content/themes/ipg/assets/images/loading-gif.gif"
            alt="Icono de carga"
          />
          <br />
          Cargandoâ€¦
        </p>
      )}
      <li className="card-grid">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </li>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
