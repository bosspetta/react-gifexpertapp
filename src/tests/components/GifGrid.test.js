import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid />', () => {
  const category = 'One Punch';

  test('Debe mostrarse correctamente', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe mostrar ítems cuando se cargan las imágenes con useFecthGifs', () => {
    const gifs = [
      {
        id: 'ABC',
        url: 'http://localhost:8080/cualquier/cosa.jpg',
        title: 'Hi React!',
      },
      {
        id: 'DEF',
        url: 'http://localhost:8080/cualquier/cosa-dos.jpg',
        title: 'Hi React again!',
      },
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={category} />);

    // expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

  });
});
