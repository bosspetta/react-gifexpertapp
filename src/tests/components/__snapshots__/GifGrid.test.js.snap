// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Debe mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    <img
      alt="Icono de carga"
      src="https://www.ipgvip.com/wp-content/themes/ipg/assets/images/loading-gif.gif"
    />
    <br />
    Cargandoâ€¦
  </p>
  <li
    className="card-grid"
  />
</Fragment>
`;
